app/main.py

```diff
@@ -1,13 +1,17 @@
 def format_linter_error(error: dict) -> dict:
-    # write your code here
-    pass
+    return {"line": error["line_number"], "column": error["column_number"],
+            "message": error["text"], "name": error["code"], "source": "flake8"
+            }
 
 
 def format_single_linter_file(file_path: str, errors: list) -> dict:
-    # write your code here
-    pass
+    return {"path": file_path,
+            "status": "passed" if len(errors) == 0 else "failed",
+            "errors": [format_linter_error(error) for error in errors]
+            }
 
 
 def format_linter_report(linter_report: dict) -> list:
-    # write your code here
-    pass
+    return [
+        format_single_linter_file(pth, er) for pth, er in linter_report.items()
```

Comment:
pth, er not clear name

Reply:
flake8 says you can't have more than 79 characters in a line, I tried to shorten it)

Reply:
> pth, er not clear name

flake8 says you can't have more than 79 characters in a line, I tried to shorten it)